### PR TITLE
Check for version compatibility on login

### PR DESCRIPTION
**Pull request by gregorjerse**
EXP-778

### CODE REVIEW

File: src/resdk/resolwe.py [hunk 1]
```diff
@@ -48,6 +50,7 @@
 DEFAULT_URL = "http://localhost:8000"
 AUTOMATIC_LOGIN_POSTFIX = "saml-auth/api-login/"
 INTERACTIVE_LOGIN_POSTFIX = "saml-auth/remote-login/"
+MINIMAL_SUPPORTED_VERSION_POSTFIX = "api/reskd_minimal_supported_version"
```
Comment (dblenkus) [inline src/resdk/resolwe.py:53]: Typo in the URL.

File: src/resdk/resolwe.py [hunk 2]
```diff
@@ -531,6 +537,20 @@ def __init__(
 
         self.logger.info(f"Successfully logged in as {username}.")
 
+    def version_check(self):
+        """Check that the server is compatible with the client."""
+        url = urljoin(self.url, MINIMAL_SUPPORTED_VERSION_POSTFIX)
+        response = requests.get(url)
+        minimal_version = version.parse(response.data["minimal_supported_version"])
+        my_version = version.parse(package_version("resdk"))
+        # TODO: probably some warning must be printed here?
```
Comment (dblenkus) [inline src/resdk/resolwe.py:546]: Remove the comment.

File: src/resdk/resolwe.py [hunk 3]
```diff
@@ -531,6 +537,20 @@ def __init__(
...
+        if my_version < minimal_version:
+            message = (
+                f"Warning: your version of ReSDK ({my_version}) is not compatible with "
```
Comment (dblenkus) [inline src/resdk/resolwe.py:549]: Add steps to update the package.

**Review by JureZmrzlikar — COMMENTED** [review]
I agree with comments from @dblenkus . Also, let's test this when the endpoint becomes operational on QA.

File: src/resdk/resolwe.py (hunk 1 above)
Comment (gregorjerse) [reply src/resdk/resolwe.py]: Fixed.

File: src/resdk/resolwe.py (hunk 2 above)
Comment (gregorjerse) [reply src/resdk/resolwe.py]: Done.

File: src/resdk/resolwe.py (hunk 3 above)
Comment (gregorjerse) [reply src/resdk/resolwe.py]: Done.

File: src/resdk/resolwe.py [hunk 4]
```diff
@@ -48,6 +50,7 @@
 DEFAULT_URL = "http://localhost:8000"
 AUTOMATIC_LOGIN_POSTFIX = "saml-auth/api-login/"
 INTERACTIVE_LOGIN_POSTFIX = "saml-auth/remote-login/"
+MINIMAL_SUPPORTED_VERSION_POSTFIX = "api/resdk_minigmal_supported_version"
```
Comment (JureZmrzlikar) [inline src/resdk/resolwe.py:53]: New typo (`minigmal`)
:-)

Comment (JureZmrzlikar) [reply src/resdk/resolwe.py:53]: Fix this typo.

File: src/resdk/resolwe.py [hunk 5]
```diff
@@ -531,6 +537,22 @@ def __init__(
 
         self.logger.info(f"Successfully logged in as {username}.")
 
+    def version_check(self):
+        """Check that the server is compatible with the client."""
+        url = urljoin(self.url, MINIMAL_SUPPORTED_VERSION_POSTFIX)
+        response = requests.get(url)
+        minimal_version = version.parse(response.data["minimal_supported_version"])
```
Comment (JureZmrzlikar) [inline src/resdk/resolwe.py:544]: This currently fails with: `AttributeError: 'Response' object has no attribute 'data'`

If you replace `response.data` with `response.json()` it works as expected.

**Review by JureZmrzlikar — CHANGES_REQUESTED** [review]
Two small fixes and we are good to go.

File: src/resdk/resolwe.py (hunk 5 above)
Comment (gregorjerse) [reply src/resdk/resolwe.py]: @JureZmrzlikar Thank you for testing. I have changed the code according to your suggestions.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**